javascript/javascript1/week4/palindrome.js

```diff
@@ -0,0 +1,31 @@
+function longestPalindrome(string) {
+  const length = string.length;
+  let result = "";
+
+  function centeredPalindrome(left, right) {
+    while (left >= 0 && right < length && string[left] === string[right]) {
+      //expand in each direction.
+      left--;
+      right++;
+    }
+
+    return string.slice(left + 1, right);
+  }
+
+  for (let i = 0; i < length - 1; i++) {
+    const oddMatch = centeredPalindrome(i, i + 1);
+
+    const evenMatch = centeredPalindrome(i, i);
+
+    // if (oddMatch.length > 1) console.log("oddMatch: " + oddMatch);
+    // if (evenMatch.length > 1) console.log("evenMatch: " + evenMatch);
+
+    if (oddMatch.length > result.length) result = oddMatch;
+    if (evenMatch.length > result.length) result = evenMatch;
+  }
```

Comment:
You could abstract this into it's own function and call it twice. Once for oddMatch and once for the evenMatch and return the result at the end of each function (instead of re-assigning result).

Abstracting functions when you can!

Reply:
I handled with other comments and thank you for commenting out, that helps me a lot. I think I will ask about this one on  next class , I didnt understand exactly